lib/ziti-tunnel-cbs/ziti_dns.c

```diff
@@ -700,14 +702,16 @@ ssize_t on_dns_req(const void *ziti_io_ctx, void *write_ctx, const void *q_packe
     dns_question *q = req->msg.question[0];
 
     if (q->type == NS_T_A || q->type == NS_T_AAAA) {
-        process_host_req(req);
+        process_host_req(req); // will send upstream if no local answer and req is recursive
     } else {
         // find domain requires normalized name
         char reqname[MAX_DNS_NAME];
         check_name(q->name, reqname, NULL);
         dns_domain_t *domain = find_domain(reqname);
         if (domain) {
-            proxy_domain_req(req, domain);
+            if (q->type == NS_T_MX || q->type == NS_T_TXT || q->type == NS_T_SRV) {
+                proxy_domain_req(req, domain);
+            }
```

Comment:
Where do we respond if it is not a supported query?

Reply:
We haven't in the past. A previous commit on this PR fixed that, but then I had second thoughts because resolvers can be touchy and I worried about changing any behavior outside of filtering the queries that are proxied.

I'll take another shot at ensuring completion, but the logic is getting a little twisted here.